### PR TITLE
feat(fe-fpm-writer): add required libraries on page addition

### PR DESCRIPTION
**Pull request by johannes-kolbe**
# Issue
#2176 

- Adds required libraries to the manifest when adding a page.
- List Report, Object Page require `sap.fe.templates`
- Custom Page requires `sap.fe.core`
- Removes check for `sap.fe.templates` prerequisite, since this will be added automatically if not existing

### CODE REVIEW

**Review by schreckstefan — DISMISSED** [review]
Took me some time to find out where the real writing takes place -at the templates, of course. Checked the snapshots, and the changes all correspond to the given description.

**Review by 815are — DISMISSED** [review]
Checked:
1. code changes easy to read/understand(small comment about 1 condition);
2. new tests are maintained;
3. changeset persists

Did not test manually - tested with running some unit tests

File: packages/fe-fpm-writer/src/common/validate.ts [hunk 1]
```diff
@@ -35,11 +36,14 @@ export function validateBasePath(basePath: string, fs?: Editor): boolean {
...
     } else {
         const manifest = fs.readJSON(manifestPath) as any;
-        if ((manifest['sap.ui5']?.dependencies?.libs?.['sap.fe.templates'] !== undefined) === false) {
-            throw new Error(
-                'Dependency sap.fe.templates is missing in the manifest.json. Fiori elements FPM requires the SAP FE libraries.'
-            );
-        }
+        const libs = manifest['sap.ui5']?.dependencies?.libs;
+        dependencies.forEach((dependency) => {
+            if ((libs?.[dependency] !== undefined) === false) {
```
[the remark this answers is not in the document]
Comment (johannes-kolbe) [reply packages/fe-fpm-writer/src/common/validate.ts]: Thanks @815are, adjusted code and updated to suggestion

**Review by 815are — APPROVED** [review]

**Review by schreckstefan — APPROVED** [review]

File: packages/fiori-elements-writer/test/__snapshots__/fpm.test.ts.snap [hunk 2]
```diff
@@ -3279,7 +3279,8 @@ EntityType=\\"com.sap.gateway.srvd.dmo.sd_travel_mduu.v0001.TravelAgencyType\\">
         \\"sap.m\\": {},
         \\"sap.ui.core\\": {},
         \\"sap.ushell\\": {},
-        \\"sap.fe.templates\\": {}
+        \\"sap.fe.templates\\": {},
+        \\"sap.fe.core\\": {}
```
Comment (devinea) [inline packages/fiori-elements-writer/test/__snapshots__/fpm.test.ts.snap:3283]: FYI This diff will be impacted by https://github.com/SAP/open-ux-tools/issues/1942 when changes re-applied.

**Review by devinea — APPROVED** [review]
FE writer snapshot changes are as expected for now. More to follow in another TBI #1942

**Pull request merged**